### DISCUSSION TITLE
Tips on setting up the environment (could be useful)

### DISCUSSION THREAD

**Discussion by Jubilee101** [pgmoneta/pgmoneta#106] [General]
After a day of trial and error, I managed to get the project running. I'm running things on a Fedora 34 aws instance, so it's slightly different from what the tutorial assumes. I'm here to share some issues I encountered that could potentially help others.
1. Aside from all the requirements mentioned in the readme that you need to install, my compiler also wanted libatomic.
2. If you install your postgres using dnf, chances are your postgres binaries is not in /usr/pgsql-10/bin as the tutorial assumes, instead just use /usr/bin/, remember to replace it in the pgmoneta.conf as well. This may help you solve the "No connection to pgmoneta on /tmp" problem
3. I also got "error while loading shared libraries: libpgmoneta.so.0: cannot open shared object" as disscussed [here](https://github.com/pgmoneta/pgmoneta/discussions/71), but you may also need to add /usr/local/lib64 because that's where my libpgmoneta.so.0 is
4. Do mind the account as denoted in brackets under every part, switch account as instructed, I ran into some nasty permission issues because of that

**Comment by jesperpedersen** [comment]
Could you create a patch for these issues ? Put stuff in the right places - `README.md`, `CONTRIBUTING.md` , `doc/tutorial` and so on.

We can discuss the precise placement in this thread as well.

Thanks !

**Reply by Jubilee101** [reply]
Sure, I'll add libatomic to requirements in `README.md`, then remind people when `/usr/pgsql-10/bin` comes up, I believe it's [here](https://github.com/pgmoneta/pgmoneta/blob/main/doc/tutorial/01_install.md#initialize-cluster) and [here](https://github.com/pgmoneta/pgmoneta/blob/main/doc/tutorial/01_install.md#create-pgmoneta-configuration). I think the libpgmoneta.so.0 error happens when you try to do `pgmoneta-admin master-key`, so maybe I'll add the solution there. Also I want to add a reminder for user account switching at the beginning of tutorial.

**Reply by jesperpedersen** [reply]
Add `libatomic` to `CONTRIBUTION.md` as well, otherwise it looks like a good proposal

**Reply by Jubilee101** [reply]
PR #109 submitted, please review it and let me know if there's problem~
